Tests/SentryTests/SentryLoggerTests.swift

```diff
@@ -628,6 +628,153 @@ final class SentryLoggerTests: XCTestCase {
         XCTAssertNil(capturedLog.attributes["device.family"])
     }
     
+    // MARK: - BeforeSendLog Callback Tests
+    
+    func testBeforeSendLogCallback_ReturnsModifiedLog() {
+        var beforeSendCalled = false
+        fixture.options.beforeSendLog = { log in
+            beforeSendCalled = true
+            
+            // Verify the mutable log has expected properties
+            XCTAssertEqual(log.level, .info)
+            XCTAssertEqual(log.body, "Original message")
+            
+            // Modify the log
+            log.body = "Modified by callback"
+            log.level = .warn
+            log.attributes["callback_modified"] = SentryLog.Attribute(boolean: true)
+            
+            return log
+        }
+        
+        sut.info("Original message")
+        
+        XCTAssertTrue(beforeSendCalled, "beforeSendLog callback should be called")
+        
+        let logs = fixture.batcher.addInvocations.invocations
+        XCTAssertEqual(logs.count, 1)
+        
+        let capturedLog = logs[0]
+        XCTAssertEqual(capturedLog.level, .warn)
+        XCTAssertEqual(capturedLog.body, "Modified by callback")
+        XCTAssertEqual(capturedLog.attributes["callback_modified"]?.value as? Bool, true)
+    }
+    
+    func testBeforeSendLogCallback_ReturnsNil_LogNotCaptured() {
+        var beforeSendCalled = false
+        fixture.options.beforeSendLog = { _ in
+            beforeSendCalled = true
+            return nil // Drop the log
+        }
+        
+        sut.error("This log should be dropped")
+        
+        XCTAssertTrue(beforeSendCalled, "beforeSendLog callback should be called")
+        
+        let logs = fixture.batcher.addInvocations.invocations
+        XCTAssertEqual(logs.count, 0, "Log should be dropped when callback returns nil")
+    }
+    
+    func testBeforeSendLogCallback_NotSet_LogCapturedUnmodified() {
+        // No beforeSendLog callback set
+        fixture.options.beforeSendLog = nil
+        
+        sut.debug("Debug message")
+        
+        let logs = fixture.batcher.addInvocations.invocations
+        XCTAssertEqual(logs.count, 1)
+        
+        let capturedLog = logs[0]
+        XCTAssertEqual(capturedLog.level, .debug)
+        XCTAssertEqual(capturedLog.body, "Debug message")
+    }
+    
+    func testBeforeSendLogCallback_MultipleLogLevels() {
+        var callbackInvocations: [(SentryLog.Level, String)] = []
+        
+        fixture.options.beforeSendLog = { log in
+            callbackInvocations.append((log.level, log.body))
+            log.attributes["processed"] = SentryLog.Attribute(boolean: true)
```

Comment:
`m`: What is the `processed` attribute and should it be configurable?

Reply:
Not sure i follow,, this is just a test for a boolean attribute. what do you mean by configurable?

Reply:
Ah I understand, it wasn't clear to me if `processed` is a special attribute like "the log was processed by the SDK", or just an example. If it's just an example ignore my comment.